frontend/src/app/forms/side-menu/side-menu.component.html

```diff
@@ -9,32 +9,77 @@ <h1>Progress</h1>
         </div>
       </div>
 
-      <ion-list lines="none" *ngFor="let menu of sideMenuList">
-        <ion-menu-toggle auto-hide="true">
+      <ion-list lines="none">
+        <ng-container
```

Comment:
Should be better if it can be done in ngFor. Whenever a new step is added we have to add ng-container here as well which can be avoided if we go Array of objcets approach but we need to bind form control valid property to isValid before opening this component. We will think and iterate it as we move forward.

Reply:
We can definitely improve this solution. But: If you create an object in ngOnInit that references an input property, then you can introduce all kinds of funny bugs. Imagine the input property changes - then this won't be reflected in the object we created in ngOnInit. So, the current solution is at least more "robust" and less error-prone.